_posts/francois_gerthoffert/2019-03-03-ZenCrepes-GitHub-Agile-Analytics-made-easy.md

```diff
@@ -0,0 +1,301 @@
+
+<figure>
+    <img src="{{site.urlimg}}francois_gerthoffert/zencrepes/jetbrains-labels.png" />
+    <figcaption>Different colors for JetBrains labels</figcaption>
+</figure>
+
+Interesting isn’t it? Look at the slight variations of red or green for “bug” and “help wanted” or the strong difference in colors for the “feature” label. Interestingly, for some reason, GitHub update from time to time the colors of the default labels, so even if you were to always stick with the default, you’d notice slight inconsistency over time.
+
+<figure>
+    <img src="{{site.urlimg}}francois_gerthoffert/zencrepes/jetbrains-labels-features.png" />
+    <figcaption>Colors for the “feature” label</figcaption>
+</figure>
+
+Actually, with the exception of 2 repositories, every single one has a different color. All of this is not of much value to an external user, but I guess if JetBrains was to want consistent colors across their repositories, ZenCrepes would allow this change to be propagated in a few minutes.
+
+# Behind the scene
+
+## Tech-Stack
+
+This app is built using [Meteor](https://www.meteor.com/) and [React](https://reactjs.org/) enhanced with [Rematch](https://rematch.gitbooks.io/rematch/#getting-started) (state management through Redux), [Material-ui](https://material-ui.com/), [Highcharts](https://www.highcharts.com/), [Nivo](https://nivo.rocks/)... 
+
+ZenCrepes pulls data about issues, labels, and milestones from GitHub and stores all of the content in [Minimongo](https://github.com/mWater/minimongo), a client-side in-memory mongodb implementation. 
+
+Communication with GitHub is a mix of their GraphQL v4 API and their REST v3 API. GraphQL is a very convenient way of querying for data, but has limited support for mutations. On the contrary, the v3 API is very complete. On slight challenge for ZenCrepes is that the model between the two is not exactly the same. To simplify things, ZenCrepes, after mutation data through the REST API will fetch data through the GraphQL API instead of parsing the response of the REST call.
+
+<figure>
+    <img src="{{site.urlimg}}francois_gerthoffert/zencrepes/graphql-explorer.png" />
+    <figcaption>I learned about GraphQL using GitHub’s GraphQL explorer (GraphiQL</figcaption>
+</figure>
+
+Issues and milestones are fetched from the latest to oldest, this allows ZenCrepes to only fetch recent changes. The first load takes the longest, subsequent calls only need to fetch the updated nodes. 
+
+From there, everything all metrics are computed in-browser.
+
+## Philosophy
+
+The entire point, from the very beginning of ZenCrepes development, was not to create another tool with a mix between secret sauce and GitHub data. I wanted to base the entire application over GitHub data model with no external data. The point was not to create another Agile tool, but to provide a solution to some of GitHub limitations when dealing with Agile project management.
```

Comment:
_"I wanted to base the entire application over **Github's** data model..."_ (possessive)